idaes/core/tests/test_control_volume_1d.py

```diff
@@ -270,6 +270,128 @@ def test_add_geometry_discretized_area():
     assert len(m.fs.cv.area) == 2
 
 
+@pytest.mark.unit
+def test_add_geometry_length_var_Var():
+    m = ConcreteModel()
+    m.fs = Flowsheet(default={"dynamic": False})
+    m.fs.pp = PhysicalParameterTestBlock()
+
+    m.fs.cv = ControlVolume1DBlock(default={
+                "property_package": m.fs.pp,
+                "transformation_method": "dae.finite_difference",
+                "transformation_scheme": "BACKWARD",
+                "finite_elements": 10})
+
+    m.fs.length = Var(initialize=4)
+
+    m.fs.cv.add_geometry(length_var=m.fs.length)
+
+    assert isinstance(m.fs.cv.length_domain, ContinuousSet)
+    assert len(m.fs.cv.length_domain) == 2
+    assert isinstance(m.fs.cv.area, Var)
+    assert len(m.fs.cv.area) == 1.0
+    assert m.fs.cv.area.value == 1.0
+    assert m.fs.cv.length is m.fs.length
+    assert isinstance(m.fs.cv.length, Var)
+    assert len(m.fs.cv.length) == 1.0
+    assert m.fs.cv.length.value == 4
+    assert m.fs.cv._flow_direction == FlowDirection.forward
```

Comment:
Minor point, but are all these tests necessary? Many just seem to assert that default behavior hasn't changed. I'm also not sure how I feel about default values being hard-coded into this test. It might be better to create two instances of `ControlVolume1DBlock`, with and without `length_var`, and assert that their variables are the same other than length. This way this test doesn't need to be updated if the defaults change.

Reply:
Not all of these are default values (although most are). In some cases however the value is assigned during the test to make sure the right value foes to the right place.